uefi/src/lib.rs

```diff
@@ -41,6 +41,8 @@
 #![feature(maybe_uninit_slice)]
 #![feature(negative_impls)]
 #![feature(ptr_metadata)]
+#![feature(allocator_api)]
+#![feature(slice_ptr_get)]
```

Comment:
If possible I'd like to avoid unconditionally adding new unstable features. See https://github.com/rust-osdev/uefi-rs/issues/452#issuecomment-1326831994 for an update on where we are with using stable Rust.

It looks like `slice_ptr_get` is just needed for `as_non_null_ptr`, which could be implemented directly easily enough: https://doc.rust-lang.org/src/core/ptr/non_null.rs.html#532-535

`allocator_api` on the other hand is necessary for this feature. The recent discussion in the [tracking issue](https://github.com/rust-lang/rust/issues/32838) leaves me unsure what the state of it is. It was marked "disposition merge" back in 2018, but it seems there are still concerns and active work on the design.

Maybe we could add a new feature to `uefi` like `unstable_alloc` and gate these additions behind that feature? Or alternatively a broader `unstable` feature that can be used for a broader set of unstable features in case there are more we want to add later.

Reply:
I see. We want to enable uefi-rs on stable, eventually. I'd prefer a generic unstable-feature, I think. Is there a convention in the ecosystem? Perhaps we should call it nightly instead of unstable?

Assuming we have that new feature, whatever the name will be, we are going to slowly transition all current nightly features into a stable state, right?

Sounds like a good plan!

Reply:
> Is there a convention in the ecosystem? Perhaps we should call it nightly instead of unstable?

I wrote a very quick and dirty [script](https://gist.github.com/nicholasbishop/64d72cd3b37524c5906b4f6158c7ec1f) to see what other crates use. It's imperfect in that it doesn't take into account recency (I could imagine that conventions have shifted over time), but the results come in at 770 crates with a `nightly` feature, 555 with an `unstable` feature.

> Assuming we have that new feature, whatever the name will be, we are going to slowly transition all current nightly features into a stable state, right? All new nightly features will land behind the new feature gate.

Yes, all new features behind the gate, and potentially transition some existing ones behind the feature gate as well.

Reply:
PR to implement the new feature gate: https://github.com/rust-osdev/uefi-rs/pull/590

Reply:
I no have idea how finish  that ... i am welder no informatik....